Week-1/assignment1/readme.md

```diff
@@ -0,0 +1,60 @@
+
+### C - Rendering engine and its use.
+
+Explanation C - The responsibility of the rendering engine is well... Rendering, that is display of the requested contents on the browser screen. Rendering engine use: - By default, the rendering engine can display HTML and XML documents and images. It can display other types of data via plug-ins or extension; for example, displaying PDF documents using a PDF viewer plug-in. However, in this chapter we will focus on the main use case: displaying HTML and images that are formatted using CS
+
+![renderingEngine](images/renderingEngine.png)
+
+### D - Parsers (HTML, CSS, etc)
+
+Explanation D - a parser is a program, usually part of a compiler, that receives input in the form of sequential source program instructions, interactive online commands, markup tags, or some other defined interface and breaks them up into parts that can be used by other components in programming. Parsers usually divide the work between two components: the lexer (sometimes called tokenizer) that is responsible for breaking the input into valid tokens, and the parser that is responsible for constructing the parse tree by analyzing the document structure according to the language syntax rules. The lexer knows how to strip irrelevant characters like white spaces and line breaks.
+
+![parser](images/parser.png)
+
+### E- Script Processors
+
+Explanation - The Script processor allows you to specify your own processor logic for a simple processor using JavaScript or Groovy. The script is entered as an option on the script processor. use Use the script processor to define some simple processing logic that cannot be achieved easily using any of the provided processors.
+
+### F - Tree construction
+
+Explanation – Rendering process has three tree construction first is DOM Tree then CSSOM Tree then Render Tree which consist of DOM and CSSOM Trees. DOM Tree – When the browser reads HTML code, whenever it encounters an HTML element like html, body, div etc., it creates a JavaScript object called a Node. Eventually, all HTML elements will be converted to JavaScript objects. After the browser has created Nodes from the HTML document, it has to create a tree-like structure of these node objects. Since our HTML elements in the HTML file are nested inside each other, the browser needs to replicate that but using Node objects it has previously created. This will help the browser efficiently render and manage the webpage throughout its lifecycle.
+
+![domtree](images/domtree.png)
+
+CSSOM Tree – After constructing the DOM, the browser reads CSS from all the sources (external, embedded, inline, user-agent, etc.) and construct a CSSOM. CSSOM stands for CSS Object Model which is a Tree Like structure just like DOM. Each node in this tree contains CSS style information that will be applied to DOM elements that it target (specified by the selector). CSSOM, however, does not contain DOM elements which can’t be printed on the screen like etc.
```

Comment:
Again... You have not mentioned what specific part of browser constructs CSS Tree.
Of course, the browser does everything... but mentioning which part of browser constructs CSS Tree will make it look authentic.

Of course, your image indirectly answers my question... but mentioning that in text will be good.